File: 53-maximum-subarray/53-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSum = Integer.MIN_VALUE;
        int maxHere = 0;
        for(int i=0;i<nums.length;i++){
            maxHere += nums[i];
            maxSum = Math.max(maxSum, maxHere);
            if(maxHere < 0){
                maxHere = 0;
            }
        }
        return maxSum;
    }
}

File: 53-maximum-subarray/53-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        max_sum = -float('inf')
        curr_max = 0
        for num in nums:
            curr_max += num
            max_sum = max(curr_max, max_sum)
            if curr_max < 0:
                curr_max = 0
        return max_sum
            
            
        